### PR TITLE
Add network e2e tests to the extended test suite

### PR DESCRIPTION
**Pull request by marun**
The changes included here should allow network e2e tests from kube to run against a multinode docker-in-docker cluster as part of the extended test suite.

### CODE REVIEW

**Comment by marun** [comment]
cc @rajatchopra

**Comment by marun** [comment]
I'm going to refactor this into an extended test so it can be run by the CI job.

**Comment by marun** [comment]
[test][extended:network]

**Comment by marun** [comment]
@danmcp Feedback welcome!  I have this working locally.  Do I have permission to trigger the test via the comment above?

**Comment by danmcp** [comment]
@marun Yes you do :)

**Comment by marun** [comment]
@danmcp The test passed (it's a trivial check so I would hope so) but the job itself failed trying to retain artifacts that are not created by the network tests.  Is that something you'd like to fix or should I dig into it?

**Comment by danmcp** [comment]
@marun Let's give @deads2k and @mfojtik a shot at fixing or suggesting an alternative.

**Comment by danmcp** [comment]
@marun And to be clear are you talking about:

Stderr from the command:

time="2015-09-01T09:12:31Z" level=fatal msg="Error response from daemon: Could not find the file /var/log/supervisor/*.log in container openshift-master" 



The errors at the bottom about artifacts are not fatal.

**Comment by marun** [comment]
@danmcp Oops, my bad.  I'll get to the bottom of that.

**Comment by marun** [comment]
[extended:network]

**Comment by danmcp** [comment]
@marun Just an FYI.  You don't need to add the extended marker again.  The test marker will combine all the extended requests across all comments and run each time you request a test.  And a test can be triggered by making a code change or by adding another [test] marker.  Which I have now just done.

**Comment by mfojtik** [comment]
@marun  the syntax is [test][extended:network]

File: hack/dind-cluster.sh [hunk 1]
```diff
@@ -179,12 +179,17 @@ function start() {
   sudo modprobe br_netfilter || true
   sudo sysctl -w net.bridge.bridge-nf-call-iptables=0
   ensure-loopback-for-dind "${NUM_NODES}"
+  if [ ! -d "${CONFIG_ROOT}" ]; then
```
Comment (mfojtik) [inline hack/dind-cluster.sh]: this is not needed, the `mkdir -p` will create the dir if doesn't exists or do nothing if it exists.

File: hack/test-extended.sh [hunk 2]
```diff
@@ -11,8 +11,6 @@ source "${OS_ROOT}/hack/util.sh"
 # Go to the top of the tree.
 cd "${OS_ROOT}"
 
-test_privileges
```
Comment (mfojtik) [inline hack/test-extended.sh]: make sure you copy this to other test runners.

File: hack/test-extended/network/run.sh [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+# This string filters the available tests.
+NETWORK_E2E_FOCUS="${NETWORK_E2E_FOCUS:-Network.*URL}"
+
+# TODO(marun) Discover these names instead of hard-coding
+CONTAINER_NAMES=( openshift-master openshift-node-1 openshift-node-2 )
+
+CLUSTER_CMD=${OS_ROOT}/hack/dind-cluster.sh
+
+function log() {
```
Comment (mfojtik) [inline hack/test-extended/network/run.sh]: make this a generic function we can re-use in other extended test?

File: hack/test-extended/network/run.sh [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+  if [ "${ENABLE_SELINUX}" = "1" ]; then
+    log-info "Re-enabling selinux enforcement"
+    setenforce 1
+  fi
+}
+trap "exit" INT TERM
+trap "cleanup-dind" EXIT
+
+# Docker-in-docker is not compatible with selinux
+if selinuxenabled; then
```
Comment (mfojtik) [inline hack/test-extended/network/run.sh]: this should be common function.

**Comment by danmcp** [comment]
@mfojtik The two commands don't need to be in the same comment and you just kicked it off again :)

**Comment by marun** [comment]
@danmcp I was checking to see if it was possible to trigger just the extended network test, but I'm guessing that's not actually possible.  Thanks for the info though, good stuff.

File: hack/test-extended.sh (hunk 2 above)
Comment (marun) [inline hack/test-extended.sh]: It's already there.

File: hack/dind-cluster.sh (hunk 1 above)
Comment (marun) [inline hack/dind-cluster.sh]: Done

File: hack/test-extended/network/run.sh (hunk 4 above)
Comment (marun) [inline hack/test-extended/network/run.sh]: Done

File: hack/test-extended/network/run.sh (hunk 3 above)
Comment (marun) [inline hack/test-extended/network/run.sh]: Done

**Comment by mfojtik** [comment]
@danmcp thanks! i thought they have to be together, everyday I learn something new ;-)

**Comment by marun** [comment]
[test]

**Comment by marun** [comment]
[extended:networking]

**Comment by marun** [comment]
@danmcp The test passes, and if there are no more comments this should be ready to merge.  However, I changed the name of the bucket from 'network' to 'networking' and the test run failed with the following error:
    [ERROR] Extended test bucket network not found

The wiki talks about the bucket string being the union of [extended:] comments, but how to remove one once it's been added?

**Comment by danmcp** [comment]
@marun I removed the existing network comments

**Comment by danmcp** [comment]
@rajatchopra Review and merge please.

**Comment by marun** [comment]
@rajatchopra I'm punting on the intra-pod test for now. The lastest patchset includes includes the more granular connectivity checks I proposed to upstream kube as well as an isolation test that runs against the multitenant plugin.

**Comment by rajatchopra** [comment]
[merge]

**Comment by marun** [comment]
@rajatchopra rebased.  re-merge?

**Comment by marun** [comment]
@rajatchopra The latest patchset fixed an issue with test logging so a re-merge appears necessary.

**Comment by danmcp** [comment]
[merge]

**Pull request merged**